flight/probability/plot_algo.py

```diff
@@ -0,0 +1,424 @@
+            for j in range(len(cell_map[0])):
+                if cell_map[i][j].is_valid:
+                    new_grid[i][j] = 1
+        return new_grid
+
+    @staticmethod
+    def get_valid_point(point, cell_map, cell_size):
+        cell_map.display()
+        point = (point[0] * cell_size, point[1] * cell_size)
+        middle = cell_size // 2
+        closest = float("inf")
+        closest_point = (point[0] + middle, point[1] + middle)
+        for i in range(cell_size):
+            for j in range(cell_size):
+                try:
+                    if cell_map[point[0] + i][point[1] + j].is_valid:
+                        dist = sqrt((middle - i) ** 2 + (middle - j) ** 2)
+                        if dist < closest:
+                            closest = dist
+                            closest_point = (point[0] + i, point[1] + j)
+                except:
+                    pass
+        return closest_point
+
+    @staticmethod
+    def __decompress_point(point: Tuple[int, int], cell_map: CellMap, cell_size: int):
+        new_x = point[1] * cell_size + (cell_size // 2)
+        new_y = point[0] * cell_size + (cell_size // 2)
+
+        if cell_map[new_y][new_x].is_valid:
+            return (new_y, new_x)
+        else:
+            return Decompressor.get_valid_point(point, cell_map, cell_size)
+
+    @staticmethod
+    def decompress_route(route: List[Tuple[int, int]], cell_map: CellMap, cell_size: int):
+        """
+        Takes the path generated using the compressed map and decompresses it
+        to the original resolution of the CellMap.
+
+        Parameters
+        ----------
+        route : List[Tuple[int, int]]
+            the compressed route
+        cell_map : CellMap
+            the original, uncompressed cell map
+        cell_size : int
+            the side length of the compressed cell
+
+        Returns
+        -------
+        uncompressed_route : List[Tuple[int, int]]
+            the uncompressed route
+        """
+        prepped_grid = Decompressor.__prep_grid(cell_map)
+        search_grid = Grid(matrix=prepped_grid)
+        finder = AStarFinder()
+        for i in range(len(route)):
+            route[i] = Decompressor.__decompress_point(route[i], cell_map, cell_size)
+        new_path = []
+        for i in range(len(route) - 1):
+            search_grid.cleanup()
+            A = search_grid.node(route[i][1], route[i][0])
+            B = search_grid.node(route[i + 1][1], route[i + 1][0])
+            A_to_B, _ = finder.find_path(A, B, search_grid)
+
+            if i == len(route) - 2:
+                new_path += A_to_B[:-1]
+            else:
+                new_path += A_to_B
+
+        return new_path
+
+
+def get_plot():
+    """
+    Gets the coordinates list for SUAS 2023
+    """
+    area = segment(rotate_shape(AIR_DROP_AREA, SUAS_2023_THETA, AIR_DROP_AREA[0]), 0.000025)
+
+    cell_map = CellMap(area, 5)
+    cell_map.display()
+    s = Searcher(cell_map, 8)
```

Comment:
Type annotations please. Not for line 398, I just got lazy and didn't want to make 2 separate comments. You already have quite enough.